### PR TITLE
Port TransitionManager from dekc.gl

### PR DESCRIPTION
**Pull request by Pessimistress**
Changes introduced in https://github.com/uber/deck.gl/pull/1154
Tests https://github.com/uber/deck.gl/pull/1183

### CODE REVIEW

File: src/components/interactive-map.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ const propTypes = Object.assign({}, StaticMap.propTypes, {
   // transition duration for viewport change
   transitionDuration: PropTypes.number,
   // function called for each transition step, can be used to perform custom transitions.
```
Comment (abmai) [inline src/components/interactive-map.js]: update comment?

File: src/utils/transition-manager.js [hunk 2]
```diff
@@ -42,97 +38,98 @@ export default class TransitionManager {
 
   // Returns current transitioned viewport.
   getViewportInTransition() {
-    return this.state.viewport;
+    return this.state.propsInTransition;
   }
 
   // Process the vewiport change, either ignore or trigger a new transiton.
```
Comment (abmai) [inline src/utils/transition-manager.js]: not related to PR change, but typo here would be nice to fix :p

File: src/utils/transition-manager.js [hunk 3]
```diff
@@ -42,97 +38,98 @@ export default class TransitionManager {
...
-      const endViewport = this.state.endViewport;
-
-      const startViewport = this.state.interruption === TRANSITION_EVENTS.SNAP_TO_END ?
-        (endViewport || currentViewport) :
-        currentViewport;
-
-      this._triggerTransition(startViewport, nextProps);
+      const startProps = Object.assign({}, currentProps,
+        this.state.interruption === TRANSITION_EVENTS.SNAP_TO_END ?
+        this.state.endProps : (this.state.propsInTransition || currentProps)
```
Comment (abmai) [inline src/utils/transition-manager.js]: nit: break to newline after the `:` to ease readability?

File: src/utils/transition-manager.js [hunk 4]
```diff
@@ -42,97 +38,98 @@ export default class TransitionManager {
...
 
   _isTransitionEnabled(props) {
-    return props.transitionDuration > 0;
+    return props.transitionDuration > 0 && props.transitionInterpolator;
   }
 
   _isUpdateDueToCurrentTransition(props) {
-    if (this.state.viewport) {
-      return areViewportsEqual(props, this.state.viewport);
+    if (this.state.propsInTransition) {
```
Comment (abmai) [inline src/utils/transition-manager.js]: use `this._isTransitionInProgress()` instead?

File: src/utils/transition-manager.js [hunk 5]
```diff
@@ -42,97 +38,98 @@ export default class TransitionManager {
...
 
-    this.props = nextProps;
+    return transitionTriggered;
   }
 
   // Helper methods
 
   _isTransitionInProgress() {
-    return this.state.viewport;
+    return this.state.propsInTransition;
```
Comment (abmai) [inline src/utils/transition-manager.js]: nit: `Boolean(this.state.propsInTransition)`

**Comment by abmai** [comment]
Woops, realized I might've been spamming you by adding single comments everywhere :p. My bad, will do a whole review next time.

**Comment by Pessimistress** [comment]
@1chandu Almost. This TransitionManager uses `MapState` to normalize viewport props instead of implementing `extractViewportProps`.

**Pull request merged**